models/sipnet/R/met2model.SIPNET.R

```diff
@@ -277,10 +277,12 @@ met2model.SIPNET <- function(in.path, in.prefix, outfolder, start_date, end_date
       }
     }
     
-    if(year.fragment){ #gets correct DOY for fragmented years 
-      doy.start <-  length(as.Date(paste0(start_year, "-01-01")):as.Date(start_date)) * (86400 / dt) + 1 #subtract to include start.date, add to exclude last half hour of day before
-      doy.end <-  length(as.Date(paste0(start_year, "-01-01")):as.Date(end_date)) * (86400 / dt)
-      doy <- doy[doy.start:doy.end]
+    if(year.fragment){ #gets correct DOY for fragmented years using start date, time since start date and end date
+      doy.seq <- as.Date(seq(from = start_date + sec[1], to = end_date, length.out = length(sec)))
```

Comment:
FYI, `seq` also has `Date` and `POSIXt` methods, so you can write code like this:

```r
seq(as.POSIXct("2010-01-01"), as.POSIXct("2010-02-01"), by = "2 days")
```

See `?seq.POSIXt` for details, but basically the `by` argument can be various strings like `2 months`, `5 days`, `3 hours`, `10 minutes`, or `5 seconds`.

Reply:
I didn't use the POSIXt method because the time between changes based on which NOAA GEFS function is used. The downscale is hourly but the regular is 6 hours so using length.out gives us the proper doy with either situation